### ISSUE TITLE
Very poor lossless compression case

### ISSUE THREAD

**Issue by greentore** [libjxl/libjxl#3109]
I encountered an image with very poor lossless compression performance in comparison to png.

base - [pingo](https://css-ig.net/pingo) optimized png: 313943
png encoded with imagemagick: 426922
`cjxl -e 3`: 958171
`cjxl -e 7`: 577244
`cjxl -e 9`: 363906
`cjxl -e 9 -g 3 -I 100 -E 3`: 232101
`cjxl -e 10`: 231444
`cwebp -z 6`: 349044
`cwebp -z 9`: 152100

**Bonus stage**
7zip .bmp to .zip compression level 5: 327647
7zip .bmp to .7z compression level 5: 154797
7zip .bmp to .7z compression level 7: 150307 👑

**Environment**
 - OS: Windows
 - cjxl/djxl version string: `JPEG XL encoder v0.9.0 6768ea8 [AVX2,SSE4,SSSE3,SSE2]`

By the way, is jxl theoretically capable of deduplicating stacked images like webp is? I deal with such images pretty often and jxl is just unable to compete even when it doesn't fail completely like this time.

![luna_genki](https://github.com/libjxl/libjxl/assets/117551249/c0ae633b-fab1-4823-adc2-0a09fdfb622b)

**Comment by jonnyawsom3** [comment]
JXL has a feature called Patches that _should_ find and reuse identical parts of the image, however in this case it's only reusing the 2 corners of the desk for some reason

**Comment by mo271** [comment]
Thanks for the report and brining up this example image.

> By the way, is jxl theoretically capable of deduplicating stacked images like webp is? I deal with such images pretty often and jxl is just unable to compete even when it doesn't fail completely like this time.

Yes, it should be possible to hand encode these images making use of the fact that the upper half is very similar to the lower half, by using storing the image as two layers, where the second references the first layer. However currently it is not possible to use cjxl to do that.

**Comment by Orum** [comment]
Yeah, I have plenty of images (50 in total, right now) where the smallest `cjxl -d 0 -e $c --streaming_input --streaming_output` (where `$c` is 1 through 10, in 0.10.0) is beaten by `cwebp`'s best `-z` level (which is not always 9).  The difference can be quite large too, with instances of the JXL being 40% larger or more than the WebP.

So, if you need any more examples, let me know and I'm happy to upload them.
